### ISSUE TITLE
Allow or force editors to provide alt text for lead images

### ISSUE THREAD

**Issue by skleinfeldt** [plone/Products.CMFPlone#2838]
When creating or editing a content item with a lead image (such as a News Item or a custom type that uses the lead image behavior), there should be a (new) field for entering alt text. The contents of this field should be put into the alt attribute of the image when the content item is viewed. The image's markup should never include a title attribute. The content item's title is displayed on the page and thus read by screen readers, putting that information in the title attribute of an image would be redundant. Alt text is the proper way to convey the image's meaning to those using screen readers. Note that when displaying the object the Title and Description will show and will be read by screen readers; the alt text should be a non-redundant description of the image itself.

If the accessibility control panel (#2834) is set to force editors to enter alt text for images, then this is a required field.

Field name: Alt Text

Help text: Briefly describe the meaning of the lead image for people using assistive technology like screen readers. Do not duplicate the object's Title or Description fields, since they will also be read by screen readers. Alt text should describe what a sighted user sees when looking at the image. This might include text the image contains, or even a description of an abstract pattern. This field should never be left blank on sites that want to be compliant with accessibility standards.

**Comment by hvelarde** [comment]
The `alt text` attribute can be empty indeed; it has to be present, but it can be empty:

https://webaim.org/techniques/alttext/

**Comment by skleinfeldt** [comment]
Same comment as https://github.com/plone/Products.CMFPlone/issues/2837#issuecomment-485865257

**Comment by hvelarde** [comment]
In this case I don't agree with you: a lead image can be anything and can be used in many ways; forcing the `alt text` attribute is no the right thing to do, IMO.

**Comment by skleinfeldt** [comment]
@hvelarde in that case the lead image alt text implementation should offer the same functionality as #2836 (if alt text is empty a comment is provided as to why), would you agree?

**Comment by iham** [comment]
Even though the alt-attr is allowed to be empty, it should be (if used) describe the image‘s content. Which it doesn‘t by the title of the object the leadimage is added to.
Which is a - better than nothing - fallback, but not a real solution in order to make accessible content as certain standards „propose“.

**Comment by jensens** [comment]
Still relevant for Plone 6. Lead image behavior has no dedicated alt text field — accessibility gap. Needs someone to implement.
